apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/BufferOptions.cs

```diff
@@ -43,51 +43,38 @@ public class BufferOptions
         private static readonly TimeSpan DefaultWaitTime = TimeSpan.FromSeconds(5);
 
         /// <summary>The type of buffer to be used.</summary>
-        public BufferType BufferType { get; private set; }
+        public readonly BufferType BufferType;
```

Comment:
Please add the `{ get; }` back in - the idea isn't that they're public fields, but public readonly properties. (You'll need to remove the readonly modifier, too.)